src/plugins/mer/meremulatordevicewidget.cpp

```diff
@@ -103,22 +192,33 @@ void MerEmulatorDeviceWidget::updatePortInputsEnabledState()
     QString message = tr("Stop emulator to unlock this field for editing.");
     if (!isStored)
         message = tr("Finish adding the device to unlock this field for editing.");
+    const QString toolTip = QString("<font color=\"red\">%1</font>");
+
+    m_ui->qmlLivePortsInfoLabel->setToolTip(toolTip.arg(message));
+    m_ui->sshPortInfoLabel->setToolTip(toolTip.arg(message));
+    m_ui->customEmulatorInfoLabel->setToolTip(toolTip.arg(message));
 
-    m_ui->qmlLivePortsInfoLabel->setToolTip(QString("<font color=\"red\">%1</font>").arg(message));
-    m_ui->sshPortInfoLabel->setToolTip(QString("<font color=\"red\">%1</font>").arg(message));
+    const bool canEditPorts = vmOff && isStored;
+    m_ui->sshPortSpinBox->setEnabled(canEditPorts);
+    m_ui->sshPortInfoLabel->setVisible(!canEditPorts);
+    m_ui->qmlLivePortsLineEdit->setEnabled(canEditPorts);
+    m_ui->qmlLivePortsInfoLabel->setVisible(!canEditPorts);
 
-    bool canEdit = vmOff && isStored;
 
-    m_ui->sshPortSpinBox->setEnabled(canEdit);
-    m_ui->sshPortInfoLabel->setVisible(!canEdit);
-    m_ui->qmlLivePortsLineEdit->setEnabled(canEdit);
-    m_ui->qmlLivePortsInfoLabel->setVisible(!canEdit);
+    const bool isCustom = MerSettings::isCustomEmulatorDeviceModel(device->deviceModel());
+    m_ui->customEmulatorInfoLabel->setVisible(isCustom && !vmOff);
+
+    m_ui->screenResolutionWidthSpinBox->setEnabled(isCustom && vmOff);
+    m_ui->screenWidthSpinBox->setEnabled(isCustom && vmOff);
+    m_ui->screenResolutionHeightSpinBox->setEnabled(isCustom && vmOff);
+    m_ui->screenHeightSpinBox->setEnabled(isCustom && vmOff);
+    m_dconfSettings->widget()->setProperty("enabled", isCustom && vmOff);
 }
 
 void MerEmulatorDeviceWidget::timeoutEditingFinished()
 {
-    Q_ASSERT(dynamic_cast<MerEmulatorDevice *>(this->device().data()) != 0);
-    MerEmulatorDevice* device = static_cast<MerEmulatorDevice*>(this->device().data());
+    auto device = this->device().dynamicCast<MerEmulatorDevice>();
+    QTC_ASSERT(device, return);
```

Comment:
Looks like a bugfix, I would prefer splitting this out to a separate commit. Similar below.